POEM_052.md

```diff
@@ -179,52 +293,40 @@ The resulting `comp` component instance would have three inputs: `x`, `y`, `z`.
 It would have three outputs `foo`, `bar`, `baz`. 
 Note that no two output names on different functions can be the same, since that would cause a name conflict in the output list. 
 
-## Providing partial derivatives
 
-Users should have access to the full `declare_partials` API, including specifying any details about finite difference or complex-step approximations and also use the `declare_coloring` component API. 
+NOTE: I would actually discourage combining multiple functions into an single component; I think it would invite name conflicts and breaks options syntax. I think a preferred signature would be `ExplicitFuncComp(func_to_comp, **comp_options_kwargs)` and naming convention should be `<function name>_comp`. So for an earlier version of `some_func` with the `num_nodes` option declared, we would do `some_func_comp = om.ExplicitFuncComp(some_fnc, num_nodes=3)` or equivalent.
```

Comment:
would you discourage it to the point of not allowing it? 

It seems possible/reasonable to me that someone might have a suite of 2 or 3 functions with highly overlapping i/o. This certainly happens for ExecComp already. 

OpenMDAO does get a performance benefit from having fewer components and fewer inputs to track and move around. 

There are some corner cases that are problematic though. If you had two functions with the same input/output name that had different metadata (e.g. size or units) that would be an error. Similarly, if you had two functions with the same option name, but different meta data that would also be an error. 

Saving those errors (which we already have the ability to check for and catch in a similar means to conflicting metadata in promoted inputs), we can just make the I/O a superset of all inputs and options from all functions. 
Is that potentially confusing, if the user adds 50 functions. Yes.  But im not sure its fundamentally a bad idea. There are lots of ways to write bad code in python... the question really is whether or not the API functionality is valuable. 

if you'd like to propose locking a func-comp to one function, Id like to see a stronger argument for why.